test/integration/regression.spec.js

```diff
@@ -54,17 +55,175 @@ describe('regressions', function () {
     });
   });
 
-  describe('issue-2286: after doesn\'t execute if test was skipped in beforeEach', function () {
-    var afterWasRun = false;
-    describe('suite with skipped test for meta test', function () {
-      beforeEach(function () { this.skip(); });
-      after(function () { afterWasRun = true; });
-      it('should be pending', function () {});
-    });
-    after('meta test', function () {
-      afterWasRun.should.be.ok();
+  describe(
+    "issue-2286: after doesn't execute if test was skipped in beforeEach",
+    function () {
+      /**
+       * Generates tests for behavior of `this.skip()`
+       * @param {string} name - Name of Runnable abort via `this.skip()`
+       * @param {Array.<boolean>} expected - For each of the Runnable types,
+       *   whether or not the Runnable should have been run.  The order is
+       *   "beforeAll" x 2, "beforeEach" x 2, "test" x 2, "afterEach" x 2, then
+       *   "afterAll" x 2.  There should be 10 items in this array.
+       * @param {string} [mode=always] - One of 'always' or 'once'
+       */
+      function testPendingRunnables (name, expected, mode) {
+        mode = mode || 'always';
+        var spies = [];
+
+        function spy (skip) {
+          function wrapped () {
+            if ((!wrapped.runCount++ || mode === 'always') && skip) {
```

Comment:
Thanks for your thoughts!  

I usually just prefer to use `!` when a falsy check is adequate.  It's a common style, so I'm not too concerned about it.

Though, some prefer explicit addition (`+`) over increment operator(s).  This is more obvious for readers of JavaScript who are, say, unfamiliar with the difference between `foo++` and `++foo` (or even `+foo`).

But, if it's opaque, I'll find another way to skin this particular cat.